opentasks-provider/src/main/java/org/dmfs/provider/tasks/model/adapters/DateTimeIterableFieldAdapter.java

```diff
@@ -0,0 +1,207 @@
+        String timeZoneId = null;
+
+        if (values != null && values.containsKey(mDateTimeListFieldName))
+        {
+            if (values.getAsString(mDateTimeListFieldName) == null)
+            {
+                // the date times are null, so we return null
+                return EmptyIterable.instance();
+            }
+            datetimeList = values.getAsString(mDateTimeListFieldName);
+        }
+        else if (cursor != null && (tsIdx = cursor.getColumnIndex(mDateTimeListFieldName)) >= 0)
+        {
+            if (cursor.isNull(tsIdx))
+            {
+                // the date times are null, so we return an empty Iterable.
+                return EmptyIterable.instance();
+            }
+            datetimeList = cursor.getString(tsIdx);
+        }
+        else
+        {
+            throw new IllegalArgumentException("Missing date time list column.");
+        }
+
+        if (mTimeZoneFieldName != null)
+        {
+            if (values != null && values.containsKey(mTimeZoneFieldName))
+            {
+                timeZoneId = values.getAsString(mTimeZoneFieldName);
+            }
+            else if (cursor != null && (tzIdx = cursor.getColumnIndex(mTimeZoneFieldName)) >= 0)
+            {
+                timeZoneId = cursor.getString(tzIdx);
+            }
+            else
+            {
+                throw new IllegalArgumentException("Missing timezone column.");
+            }
+        }
+
+        // create a new TimeZone for the given time zone string
+        TimeZone timeZone = timeZoneId == null ? null : TimeZone.getTimeZone(timeZoneId);
+
+        return new DateTimeList(timeZone, datetimeList);
+    }
+
+
+    @Override
+    public void setIn(ContentValues values, Iterable<DateTime> value)
+    {
+        if (value != null)
+        {
+            String stringValue = new Reduced<>(
+                    new StringBuilder(256),
+                    StringBuilder::append,
+                    new Alternating<>(
+                            new Mapped<>(
+                                    DateTime::toString,
+                                    new Mapped<>(dt -> dt.isFloating() ? dt : dt.shiftTimeZone(DateTime.UTC), value)),
+                            new ConstantGenerator<>(","))).value().toString();
+            values.put(mDateTimeListFieldName, stringValue.isEmpty() ? null : stringValue);
+        }
+        else
+        {
+            values.put(mDateTimeListFieldName, (String) null);
+        }
+    }
+
+
+    private final class DateTimeList extends DelegatingIterable<DateTime>
+    {
+
+        public DateTimeList(TimeZone timeZone, String dateTimeList)
+        {
+            super(new Mapped<>(
+                    datetime -> !datetime.isFloating() && timeZone != null ? datetime.shiftTimeZone(timeZone) : datetime,
+                    new Mapped<CharSequence, DateTime>(
+                            charSequence -> DateTime.parse(timeZone, charSequence.toString()),
```

Comment:
Why is it needed to shift the timezone, doesn't `DateTime.parse()` set it up correctly?

Reply:
If the String ends with a `Z` the `parse` method will always return a `UTC` `DateTime`, no matter which time zone was passed.

Reply:
I see. It's seems a bit difficult and thus error prone to keep track of the proper ways of `DateTime` conversions.
How about creating more elemental `Single`s for them and collecting them in a dedicated package?
The `Single<DateTime>` which could be extracted from here would take a `TimeZone` and a string timestamp.
For me that would be a much clearer situation, to have that few classes defined somewhere which are for sure right, and can be re-used. The goal would be to not have to shift, swap timezone and things like that anywhere else in code.

This could be a separate ticket of course. There is already some work in progress related to this on my side as well.

Reply:
Right, we can certainly do that. I was also planning on reworking the DateTime class into something more SRP-ish.